src/Sylius/Bundle/CoreBundle/Test/Factory/TestUserFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Test\Factory;
+
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class TestUserFactory implements TestUserFactoryInterface
+{
+    const DEFAULT_USER_EMAIL = 'john.doe@example.com';
+    const DEFAULT_USER_FIRST_NAME = 'John';
+    const DEFAULT_USER_LAST_NAME = 'Doe';
+    const DEFAULT_USER_PASSWORD = 'password123';
+
+    /**
+     * @var FactoryInterface
+     */
+    private $customerFactory;
+
+    /**
+     * @var FactoryInterface
+     */
+    private $userFactory;
+
+    /**
+     * @param FactoryInterface $customerFactory
+     * @param FactoryInterface $userFactory
+     */
+    public function __construct(FactoryInterface $customerFactory, FactoryInterface $userFactory)
+    {
+        $this->customerFactory = $customerFactory;
+        $this->userFactory = $userFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create($email, $password, $firstName = self::DEFAULT_USER_FIRST_NAME, $lastName = self::DEFAULT_USER_LAST_NAME)
+    {
+        $customer = $this->customerFactory->createNew();
+
+        $customer->setFirstName($firstName);
+        $customer->setLastName($lastName);
+
+        $user = $this->userFactory->createNew();
+
+        $user->setCustomer($customer);
+        $user->setUsername($email);
+        $user->setEmail($email);
+        $user->setPlainPassword($password);
+        $user->enable();
```

Comment:
I just wanted to point the same thing ;) https://github.com/Sylius/Sylius/blob/master/src/Sylius/Component/User/Model/User.php#L123